### PR TITLE
Adds description to project api struct and model

### PR DESCRIPTION
**Pull request by daniellemaxwell**
### Scope of changes

Adds `description` to `Project` API struct and model. Beacon users will have an option to include a description when creating a project. If a description is provided, it will be included in the `Project Detail` tooltip.

Links to the Figma designs are below for reference:
https://www.figma.com/file/PvL23fm6W36Zd88O1XVyIo/Beacon-DX-UI?node-id=4631-25864&t=T9NuNKp8hwnJhaNi-4

https://www.figma.com/file/PvL23fm6W36Zd88O1XVyIo/Beacon-DX-UI?node-id=5374-14608&t=T9NuNKp8hwnJhaNi-4

Fixes SC-16585

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/tenant/projects.go [hunk 1]
```diff
@@ -156,6 +156,10 @@ func (s *Server) TenantProjectCreate(c *gin.Context) {
 		Name:     project.Name,
 	}
 
+	if project.Description != "" {
```
Comment (daniellemaxwell) [inline pkg/tenant/projects.go:159]: The Figma design doesn't include a max word count, but I'm wondering if we should add a check for that.

Comment (daniellemaxwell) [reply pkg/tenant/projects.go:159]: Never mind. In SC-15723, the max length for the description is listed as 2k characters.

**Review by pdeziel — COMMENTED** [review]
Looks great, I agree that the description should be optional. I think we should add the length check as you suggested to prevent the user from being able to add arbitrarily long descriptions, so I didn't mark this as approved yet.

File: pkg/tenant/db/projects.go [hunk 2]
```diff
@@ -72,10 +73,11 @@ func (p *Project) Validate() error {
 // Convert the model to an API response.
 func (p *Project) ToAPI() *api.Project {
 	return &api.Project{
-		ID:       p.ID.String(),
-		Name:     p.Name,
-		Created:  TimeToString(p.Created),
-		Modified: TimeToString(p.Modified),
+		ID:          p.ID.String(),
+		Name:        p.Name,
+		Description: p.Description,
```
[the remark this answers is not in the document]
Comment (pdeziel) [reply pkg/tenant/db/projects.go:78]: Can we also add an assertion in the project detail test to make sure that the description is being returned?

[the remark this answers is not in the document]
Comment (daniellemaxwell) [reply pkg/tenant/db/projects.go:82]: Yes, that makes sense as the description will not need to be returned on list.

File: pkg/tenant/projects.go [hunk 3]
```diff
@@ -424,6 +433,15 @@ func (s *Server) ProjectUpdate(c *gin.Context) {
 	// Update all user provided fields
 	p.Name = project.Name
 
+	if project.Description != "" {
```
Comment (daniellemaxwell) [inline pkg/tenant/projects.go:436]: I added this after realizing that user's may want to edit their project description.

File: pkg/tenant/db/projects_test.go [hunk 4]
```diff
@@ -67,8 +69,13 @@ func TestProjectValidate(t *testing.T) {
 	project.Name = " "
 	require.ErrorIs(t, project.Validate(), db.ErrMissingProjectName, "expected missing name error")
 
-	// Test valid project
+	// Test description that's too long
 	project.Name = "Hello World"
+	project.Description = strings.Repeat("a", 2001)
```
Comment (pdeziel) [inline pkg/tenant/db/projects_test.go:74]: I did not know about `strings.Repeat`, nice test hack!

File: pkg/tenant/db/projects.go [hunk 5]
```diff
@@ -66,16 +67,21 @@ func (p *Project) Validate() error {
 		return ErrMissingProjectName
 	}
 
+	if len(p.Description) > 2000 {
+		return ErrProjectDescriptionTooLong
```
Comment (pdeziel) [inline pkg/tenant/db/projects.go:71]: Can you make sure that this constraint is mentioned in the frontend story for project create?

File: pkg/tenant/projects.go [hunk 6]
```diff
@@ -156,6 +156,15 @@ func (s *Server) TenantProjectCreate(c *gin.Context) {
 		Name:     project.Name,
 	}
 
+	if project.Description != "" {
+		if len(project.Description) > 2000 {
```
Comment (pdeziel) [inline pkg/tenant/projects.go:160]: We'll probably want to make this a constant so we can share it between api and db validation.

File: pkg/tenant/projects.go (hunk 3 above)
Comment (pdeziel) [reply pkg/tenant/projects.go:436]: Agreed, good call!

**Review by pdeziel — APPROVED** [review]
Looks good, I had one suggestion about changing the max length to a constant so we don't have to update it in three places, but other than that looks good!

File: pkg/tenant/db/projects.go (hunk 5 above)
Comment (daniellemaxwell) [reply pkg/tenant/db/projects.go:74]: Edwin included the constraint when he created the FE story.

**Pull request merged**